src/platform_linux/platform.c

```diff
@@ -211,11 +249,28 @@ platform_thread_id_self()
 platform_status
 platform_mutex_init(platform_mutex    *lock,
                     platform_module_id UNUSED_PARAM(module_id),
-                    platform_heap_id   UNUSED_PARAM(heap_id))
+                    platform_heap_id   heap_id)
 {
-   int ret     = pthread_mutex_init(&lock->mutex, NULL);
-   lock->owner = INVALID_TID;
-   return CONST_STATUS(ret);
+   platform_status status;
+   // Init mutex so it can be shared between processes, if so configured
+   pthread_mutexattr_t mattr;
+   pthread_mutexattr_init(&mattr);
+   status.r = pthread_mutexattr_setpshared(&mattr, PTHREAD_PROCESS_SHARED);
+   if (!SUCCESS(status)) {
+      return status;
+   }
+
+   // clang-format off
+   status.r = pthread_mutex_init(&lock->mutex,
+                                 ((heap_id == PROCESS_PRIVATE_HEAP_ID)
+                                    ? NULL : &mattr));
```

Comment:
Here onwards, collection of changes to migrate thread-specific synchronization logic to also work in multi-process model.